client/src/main/java/se/fortnox/reactivewizard/client/HttpClient.java

```diff
@@ -177,28 +177,22 @@ public Object invoke(Object proxy, Method method, Object[] arguments) {
 
         Mono<RwHttpClientResponse> response = request.submit(rxClient, request);
 
-        Publisher<?> publisher = null;
         AtomicReference<HttpClientResponse> rawResponse = new AtomicReference<>();
-        if (expectsByteArrayResponse(method)) {
-            publisher = response.flatMap(rwHttpClientResponse -> {
-                rawResponse.set(rwHttpClientResponse.getHttpClientResponse());
-                if (rwHttpClientResponse.getHttpClientResponse().status().code() >= 400) {
-                    return Mono.from(collector.collectString(rwHttpClientResponse.getContent()))
-                        .map(data -> handleError(request, rwHttpClientResponse.getHttpClientResponse(), data).getBytes());
-                }
-                return Mono.from(collector.collectBytes(rwHttpClientResponse.getContent()));
-            });
-        } else {
-            publisher = response.flatMap(rwHttpClientResponse -> {
-                rawResponse.set(rwHttpClientResponse.getHttpClientResponse());
-                return parseResponse(method, request, rwHttpClientResponse);
-            });
-        }
+
+        Publisher<?> publisher = response.flatMapMany(rwHttpClientResponse -> {
+            rawResponse.set(rwHttpClientResponse.getHttpClientResponse());
+
+            if (rwHttpClientResponse.getHttpClientResponse().status().code() >= 400) {
+                return parseErrorResponse(method, request, rwHttpClientResponse);
+            }
+
+            return parseResponse(method, request, rwHttpClientResponse);
+        });
+
         publisher = measure(request, publisher);
 
-        //End of publisher
         Flux<?> flux = Flux.from(publisher);
-        flux = flux.timeout(Duration.of(timeout, timeoutUnit));
+        flux      = flux.timeout(Duration.of(timeout, timeoutUnit));
```

Comment:
Not related to your code change but putting this duration into a field instead would have been nice